pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -303,7 +303,7 @@ private <C extends IndexConfig> ColumnConfigDeserializer<C> getDeserializer(Inde
         deserializer = IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig);
       } else if (_segmentTier == null) {
         deserializer =
-            IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig).withFallbackAlternative(stdDeserializer);
+            stdDeserializer.withFallbackAlternative(IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig));
```

Comment:
I think this is the change @Jackie-Jiang was worried about. And I think it makes sense to apply the change. Before this PR, our index config test verified the behavior in the ideal scenario where `IndexingLoadingConfig` was not used. But this PR adds a new check to verify that semantics are conserved when `IndexLoadingConfig` is used... and it proves there are issues right now in this case.

I'm not 100% sure if this is the correct solution, but in indexes like ForwardIndex that are enabled by default, if we prioritizes `IndexConfigDeserializer.fromMap(table -> fromIndexLoadingConfig)` instead of `stdDeserializer`, the `stdDeserializer` is always ignored which is clearly incorrect.

IMHO the correct solution would be to remove `IndexingLoadingConfig`, as we planned during `index-spi`.